src/include/migraphx/run_loop.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_RUN_LOOP_HPP
+#define MIGRAPHX_GUARD_RTGLIB_RUN_LOOP_HPP
+
+#include <string>
+#include <migraphx/instruction_ref.hpp>
+#include <migraphx/shape.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/context.hpp>
+#include <migraphx/module.hpp>
+#include <migraphx/config.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+template <class LoopModel, class T>
+argument run_loop(const LoopModel& model,
+                  T& ctx,
+                  std::vector<argument> args,
+                  const std::vector<module_ref>& mods,
+                  const std::function<std::vector<argument>(
+                      module_ref&, const std::unordered_map<std::string, argument>&)>& run)
+{
+    auto get_output_index = [](const std::string& name) {
+        std::string out_prefix = "#output_";
+        auto loc               = name.find(out_prefix);
+        if(loc != std::string::npos)
+        {
+            int index = std::stoi(name.substr(loc + out_prefix.size()));
+            return index;
```

Comment:
It says there is no coverage here. There are other places where there is no coverage in this file we should probably add unit tests for. I assume there is no coverage because output params are only used by the GPU backend. Perhaps, we can create mock loop_model and some fake output params to check with the unit tests.

Reply:
Yes, this part of code only applied to the GPU implementation.